### ISSUE TITLE
Provide basic information how to setup/integrate this bundle into an existing Sylius installation

### ISSUE THREAD

**Issue by Headd2k** [MichaelMackus/SyliusSubscriptionBundle#1]
hi!

first of all i have to say "thank you!" for this bundle. looks like it provides almost everything i want/need in my own sylius shop.

unfortunately i'm pretty new to sylius and symfony so im curious how to integrate this bundle to my existing app. 

is the any chance to get a very basic setup guide how to integrate this bundle correctly?
- add "foo" to your composer json
- activate bundle by adding "foo" to your app kernel
- extend model "foo" with properties a, b, c and relation "bar" ...
- setup cron job to execute command "foo"
- extend backend menue ...
- extend cart view ...
- create controller "foo"
- configure route "foo"
- ...

a quick overview of what is provided by the bundle and what i have to setup additionally would also be really helpful for a newbie like me... :)

thanks in advance!

kai

**Comment by MichaelMackus** [comment]
@Headd2k thanks for the input - I'll try to get something together when I find some time. 

There's still some things that need to be implemented in this repo though, like the actual payment processing (CIM or whatever else you're using). But the interface & cron commands should be largely there.

**Comment by Headd2k** [comment]
@MichaelMackus thank you very much for your reply! would be great to get some additional input how to use this bundle. thanks in advance!

**Comment by Headd2k** [comment]
finally found some of my "missing links" in the sylius-subscription branch of @Richtermeister 

https://github.com/Richtermeister/Sylius/tree/subscription-bundle

here i found the new "subscription" entity and some controllers i was searching for. thanks again for your awesome work!

**Comment by Richtermeister** [comment]
@Headd2k Yeah, the reason this is all a bit raw right now is that we had to implement this functionality in one of our applications, and couldn't wait for Sylius to approve and merge our code in. Overall it is our intention to contribute the subscription functionality to Sylius, but we're not quite done implementing it ourselves.

**Comment by zacbrac** [comment]
Hey, just curious if you guys could provide any input in how to bypass this error I'm having
```
[Doctrine\DBAL\DBALException]                                                                                                               
  Unknown column type "sylius_date_interval" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::  
  addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database i  
  ntrospection then you might have forgot to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMappi  
  ng() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache   
  or forgot some mapping information.
```
Any help would be greatly appreciated!

**Comment by MichaelMackus** [comment]
@zacbrac you have to add the custom DateInterval Doctrine type extension to your doctrine config in your config.yaml. 

For example, something like:

```yaml
doctrine:
    dbal:
        types:
            sylius_date_interval:  Sylius\Bundle\SubscriptionBundle\Doctrine\Type\DateIntervalType
```

It looks like I didn't include the DateIntervalType in the initial commit for this repo - I've added that & pushed to the master branch. 

Hopefully that helps. Ideally in the future we won't need this configuration if we can find another bundle/library to handle this implementation.

**Comment by zacbrac** [comment]
@MichaelMackus Yes, thank you so much that helps a lot! If I end up getting it working I would be happy to write the basic info on setting it up. Would that be alright with you?

**Comment by MichaelMackus** [comment]
Sure! Any pull requests are welcome. I haven't gotten around to it myself
but it's definitely been on my mind.

**Comment by Vinit-Koshti** [comment]
@Headd2k @MichaelMackus @Richtermeister  How to add subscribable attribute to the product item from admin dashboard? though I have got (My Subscriptions) on the left menu.
![screenshot from 2018-12-04 16-40-04](https://user-images.githubusercontent.com/16951071/49439485-241a2780-f7e7-11e8-96d9-b2e2b92ac243.png)
![screenshot from 2018-12-04 16-41-02](https://user-images.githubusercontent.com/16951071/49439487-241a2780-f7e7-11e8-82b6-c790294cea16.png)
